### PR TITLE
vocabbuilder.koplugin: support undo word study

### LINKED ISSUE #9527: FR: Vocabulary builder: reset progress one step back
**Does your feature request involve difficulty completing a task? Please describe.**
Sometimes, using the Vocabulary builder, we accidentally click on an option ("Got it" or "Forgot"). It would be convenient to have an option to return to the immediately previous state. Now there is the option "Reset progress", which cancels all progress in learning the word.

**Describe the solution you'd like**
If possible, add a button with the option "reset last selection" or something similar.

Dear @weijiuqiao, what do you think? Could it be possible?
Thanks!

### PR DESCRIPTION
**Pull request by weijiuqiao**
Adds button to undo last study operation (got it/forgot) in more (...) per fr #9527
Closes #9527.

- - -
This change is [Reviewable](https://reviewable.io/reviews/koreader/koreader/9528)

### CODE REVIEW

File: plugins/vocabbuilder.koplugin/main.lua [hunk 1]
```diff
@@ -398,9 +398,20 @@ function WordInfoDialog:init()
         end
     }
 
+    local buttons = {{reset_button, remove_button}}
+    if self.show_parent.item.last_due_time then
+        table.insert(buttons, {{
+            text = _("Undo study"),
```
Comment (Frenzie) [inline plugins/vocabbuilder.koplugin/main.lua:404]: I'm not entirely happy with that. Would a plain undo be unclear in context?

Comment (weijiuqiao) [reply plugins/vocabbuilder.koplugin/main.lua:404]: I believe yes. The undo button is in the `more` popup widget and each word has its own. Between one pair of `do` and `undo`, a user could have done many other things which could make a plain `undo`'s meaning less clear.

Comment (Frenzie) [reply plugins/vocabbuilder.koplugin/main.lua:404]: Undo study status?

Comment (weijiuqiao) [reply plugins/vocabbuilder.koplugin/main.lua]: Done.

**Pull request merged**